mitodl/ocw-studio
# Implement select widget

#### Pre-Flight checklist


#### What are the relevant tickets?
Fixes #140 

#### What's this PR do?
Implements the select widget and some frontend validation for the `min` and `max` field settings.

#### How should this be manually tested?

 - Edit `site-config-override.yml` and add these two lines to the metadata fields list (should be the very last lines of the file at the moment):

        - { "label": "Tags", "default": ["Design"], "max": 3, "min": 1, "multiple": true, "name": "tags", "options": ["Design", "UX", "Dev"], "widget": "select" }
        - { label: "Align Content", name: "align", widget: "select", options: ["left", "center", "right"]}
        
 - Go to `/admin/websites/websitestarter/1/change/` or a similar URL for your WebsiteStarter. Update the config textbox to include the new line you added above. If you are using the default WebsiteStarter called "OCW Course Hugo Starter" you can just copy the paste the contents from `site-config-override.yml` minus the `collections:` line at the top.
 - Go to a page for a site which uses that starter. Click on "Site Metadata" on the sidebar then click "Add Site Metadata". You should see a field called "Tags", and it should have a value "Design" set as a default option.
 - Update the items in the Tags select box. You should be able to add and remove items. Click Save and they should be saved correctly.
 - You should also be able to update the Align Content field, though you should not be able to add or remove multiple items.
 - Go back to the Site Metadata page and click Edit. You should see the fields you updated with the new values.
 - Edit your metadata entry again and remove all selected items from the "Tags" field. Click Save. You should see an error message saying that there must be at least 1 item for that field.


#### Screenshots (if appropriate)
![Screenshot from 2021-03-24 16-57-57](https://user-images.githubusercontent.com/863262/112382351-1fcced80-8cc2-11eb-89bf-a84c6c9d6ef9.png)
